okhttp/src/main/java/okhttp3/internal/http1/Http1Codec.java

```diff
@@ -74,12 +77,15 @@
   private static final int STATE_OPEN_RESPONSE_BODY = 4;
   private static final int STATE_READING_RESPONSE_BODY = 5;
   private static final int STATE_CLOSED = 6;
+  private static final int STATE_CANCEL = 7;
 
   /** The client that configures this stream. May be null for HTTPS proxy tunnels. */
   private final OkHttpClient client;
   /** The stream allocation that owns this stream. May be null for HTTPS proxy tunnels. */
   private final StreamAllocation streamAllocation;
 
+  private final Http1ConnectionTimeout connectionTimeout;
```

Comment:
Should this be called `connectionDeadline` or something with deadline in it? I think there is a difference between a timeout and a deadline.